lib/src/main/kotlin/trees/AVLTree.kt

```diff
@@ -0,0 +1,198 @@
+package trees
+
+import nodes.AVLNode
+
+import kotlin.math.max
+
+class AVLTree<K : Comparable<K>, V> : Tree<K, V, AVLNode<K, V>>() {
+
+    private fun height(node: AVLNode<K, V>?): Int {
+        if (node == null) {
+            return 0
+        }
+        return node.height
+    }
```

Comment:
здесь бы лучше зашёл безопасный вызов и элвис оператор
```suggestion
    private fun height(node: AVLNode<K, V>?): Int {
        return node?.height ?: 0
    }
```